llvm/lib/Debuginfod/Debuginfod.cpp

```diff
@@ -62,15 +66,23 @@ bool canUseDebuginfod() {
 }
 
 SmallVector<StringRef> getDefaultDebuginfodUrls() {
-  const char *DebuginfodUrlsEnv = std::getenv("DEBUGINFOD_URLS");
-  if (DebuginfodUrlsEnv == nullptr)
-    return SmallVector<StringRef>();
-
-  SmallVector<StringRef> DebuginfodUrls;
-  StringRef(DebuginfodUrlsEnv).split(DebuginfodUrls, " ");
+  if (!DebuginfodUrlsSet) {
+    // Only read from the environment variable if the user hasn't already
+    // set the value
+    if (const char *DebuginfodUrlsEnv = std::getenv("DEBUGINFOD_URLS"))
+      StringRef(DebuginfodUrlsEnv).split(DebuginfodUrls, " ", -1, false);
+    DebuginfodUrlsSet = true;
```

Comment:
If we use std::optional, this can be:
```
DebuginfodUrls = SmallVector<StringRef>();
if (const char *DebuginfodUrlsEnv = std::getenv("DEBUGINFOD_URLS"))
  StringRef(DebuginfodUrlsEnv).split(DebuginfodUrls.value(), " ", -1, false);
```

Reply:
Much better